### PR TITLE
Add imagenet learning schedule

### PR DESCRIPTION
**Pull request by zcain117**
This is option #1 for the learning rate schedule that gets resnet50 to >=76% accuracy on TPU.

It is simple but not very flexible and not ideal for assigning different learning schedules for models outside of resnet50.

Option #2 is to subclass torch.optim.lr_scheduler: https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html

This will be more code total and require some changes to `xla_model.optimizer_step`: https://github.com/pytorch/xla/blob/master/torch_xla_py/xla_model.py#L372

I have not tested option #2 but there aren't any huge blockers that I can think of. I can put together a PR for option #2 so people can compare if we think it's worthwhile

### CODE REVIEW

**Comment by zcain117** [comment]
> That is A LOT of code in the core loop function.
> Doesn't pytorch have readt to use LR scheduling APIs?

They offer a few, but none that have both a warmup period and a decay. This one is close but doesn't have warmup: https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.StepLR

I wasn't able to get to 76% accuracy without a warmup period.

Whether I add a new optimizer or use a different one from [their offerings](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html) we'll still need to modify [xm.optimizer_step](https://github.com/pytorch/xla/blob/master/torch_xla_py/xla_model.py#L372) to supply epoch, step, and steps_per_epoch to the optimizer

**Comment by dlibenzi** [comment]
> > That is A LOT of code in the core loop function.
> > Doesn't pytorch have readt to use LR scheduling APIs?
> 
> They offer a few, but none that have both a warmup period and a decay. This one is close but doesn't have warmup: https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.StepLR
> 
> I wasn't able to get to 76% accuracy without a warmup period.
> 
> Whether I add a new optimizer or use a different one from [their offerings](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html) we'll still need to modify [xm.optimizer_step](https://github.com/pytorch/xla/blob/master/torch_xla_py/xla_model.py#L372) to supply epoch, step, and steps_per_epoch to the optimizer

For the optimizer issue ... https://github.com/pytorch/xla/pull/1005

My concern is that we want our core loop to be as neat as possible, so if our optimizer does something special others do not do (this is a common problem, so it seems strange to me), we need to try to make this code as tight as possible reducing core loop clutter.

**Comment by zcain117** [comment]
> > > That is A LOT of code in the core loop function.
> > > Doesn't pytorch have readt to use LR scheduling APIs?
> > 
> > 
> > They offer a few, but none that have both a warmup period and a decay. This one is close but doesn't have warmup: https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.StepLR
> > I wasn't able to get to 76% accuracy without a warmup period.
> > Whether I add a new optimizer or use a different one from [their offerings](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html) we'll still need to modify [xm.optimizer_step](https://github.com/pytorch/xla/blob/master/torch_xla_py/xla_model.py#L372) to supply epoch, step, and steps_per_epoch to the optimizer
> 
> For the optimizer issue ... #1005
> 
> My concern is that we want our core loop to be as neat as possible, so if our optimizer does something special others do not do (this is a common problem, so it seems strange to me), we need to try to make this code as tight as possible reducing core loop clutter.

Sounds good - I'll make the optimizer subclass version

**Comment by zcain117** [comment]
> > > > That is A LOT of code in the core loop function.
> > > > Doesn't pytorch have readt to use LR scheduling APIs?
> > > 
> > > 
> > > They offer a few, but none that have both a warmup period and a decay. This one is close but doesn't have warmup: https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.StepLR
> > > I wasn't able to get to 76% accuracy without a warmup period.
> > > Whether I add a new optimizer or use a different one from [their offerings](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html) we'll still need to modify [xm.optimizer_step](https://github.com/pytorch/xla/blob/master/torch_xla_py/xla_model.py#L372) to supply epoch, step, and steps_per_epoch to the optimizer
> > 
> > 
> > For the optimizer issue ... #1005
> > My concern is that we want our core loop to be as neat as possible, so if our optimizer does something special others do not do (this is a common problem, so it seems strange to me), we need to try to make this code as tight as possible reducing core loop clutter.
> 
> Sounds good - I'll make the optimizer subclass version

Added 2 other options:
[scheduler-style](https://github.com/pytorch/xla/pull/1012)
[optimizer-style](https://github.com/pytorch/xla/pull/1011)
